staging/src/k8s.io/kubectl/pkg/describe/util.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package describe
+
+import v1 "k8s.io/api/core/v1"
+
+// IsPodPhaseTerminal returns true if the pod's phase is terminal.
+// copied from k8s.io/kubernetes/pkg/api/v1/pod
+func isPodPhaseTerminal(phase v1.PodPhase) bool {
```

Comment:
I think, we don't need this new package. It is only used by one place in `describe` and the logic is pretty simple. What about;

```go
if pod.DeletionTimestamp != nil && phase != v1.PodFailed && phase != v1.PodSucceeded {
```

Reply:
The original code is `if pod.DeletionTimestamp != nil && (pod.Status.Phase == corev1.PodRunning || pod.Status.Phase == corev1.PodPending) ` and @mimowo think it is not simple enougth. You can check that in comments.

Reply:
I was rather exploring other options, I like the proposal `&& phase != v1.PodFailed && phase != v1.PodSucceeded`. We can also employ it in the other place.

Reply:
As `printers` can access `"k8s.io/kubernetes/pkg/api/v1/pod"`, it is better to use it as is.

Reply:
> As `printers` can access `"k8s.io/kubernetes/pkg/api/v1/pod"`, it is better to use it as is.

 CI `pull-kubernetes-dependencies` will failed.  `"k8s.io/kubernetes/pkg/api/v1/pod"` is only used under `/pkg/` dir. There is also a internal package in `staging/src/k8s.io/endpointslice/util/controller_utils.go`

Reply:
I'm just saying that the current code changes in this PR for `pkg/printers/internalversion/printers.go` is ok

Reply:
@lowang-bh once we remove this staging/src/k8s.io/kubectl/pkg/describe/util.go file and use it inline, I think we are ready to go?

Reply:
done.